scRNA-seq-advanced/setup/intro-diff-expression-exploration.Rmd

```diff
@@ -0,0 +1,800 @@
+sample_metadata <- readr::read_tsv(sample_metadata_file)
+```
+Let's take the sample metadata file and add in the metadata to the `colData` of the combined SCE object. 
+While doing this we can also do some modifying of the columns in the `colData`, such as creating a `celltype_broad` and `celltype_fine` column to use for plotting.  
+
+```{r}
+# collapse cell types 
+coldata_df <- colData(combined_sce) %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column("cell_id") %>%
+  # remove any of the subtypes to create new cell type column 
+  dplyr::mutate(celltype_broad = stringr::str_remove(celltype, "-[ABCD]$"),
+                celltype_fine = celltype) %>%
+  # merge with sample metadata 
+  dplyr::left_join(sample_metadata, by = c("batch" = "library_id")) %>%
+    # simplify subdiagnosis for later
+  dplyr::mutate(diagnosis_groups = factor(
+    dplyr::case_when(
+      subdiagnosis == "Alveolar rhabdomyosarcoma" ~ "ARMS",
+      subdiagnosis == "Embryonal rhabdomyosarcoma" ~ "ERMS"
+    )))
+
+# add back to sce 
+colData(combined_sce) <- DataFrame(coldata_df, row.names = coldata_df$cell_id)
+```
+
+Let's just do some initial exploration of the integrated dataset that we are working with. 
+
+```{r}
+# UMAP to show cell type and integration  
+scater::plotReducedDim(combined_sce,
+                       dimred = "fastMNN_UMAP",
+                       colour_by = "batch",
+                       point_size= 0.5,
+                       point_alpha = 0.2)
+
+```
+
+```{r}
+# show cell types in the integrated dataset 
+scater::plotReducedDim(combined_sce,
+                       dimred = "fastMNN_UMAP",
+                       colour_by = "celltype_broad",
+                       point_size= 0.5, 
+                       point_alpha = 0.4)
+
+# facet the UMAP by sample to show the distribution of cell types across each sample
+scater::plotReducedDim(combined_sce,
+                       dimred = "fastMNN_UMAP",
+                       colour_by = "celltype_broad",
+                       point_size= 0.5, 
+                       point_alpha = 0.4,
+                       other_fields = "batch") +
+  facet_wrap(~ batch, nrow = 3)
+```
+
+Looking at this plot lets us know which cell types belong to which sample and can help us pick which libraries we might want to consider for differential expression.
+From this we see that 6 of the libraries show a split between `Tumor_Mesoderm`, `Tumor_Myoblast` and `Tumor_Myocyte`.
+3 libraries are mostly `Tumor` and they probably weren't able to be classified into sub types. 
+Then 1 library has what appears to be all `NA`. 
+Let's just look at the libraries that have myoblast, mesoderm, and myocytes and see if we can do some differential expression between those libraries. 
+
+## Set up of dataset to use for integration
+
+```{r}
+# define libraries to keep
+scpca_library_ids <- c(
+  "SCPCL000479",
+  "SCPCL000480",
+  "SCPCL000481",
+  "SCPCL000484",
+  "SCPCL000488",
+  "SCPCL000491"
```

Comment:
Just noting that these are NOT also in the integration instruction notebook. Might want to add a small comment about that for future reference.